frontend/src/components/Post.js

```diff
@@ -3,25 +3,71 @@ import Comment from './Comment'
 import AddComment from './AddComment'
 import './Post.css'
 
+
 const Post = props => {
   const [replyOpen, setReplyOpen] = React.useState(false)
+  const originalVotes = props.post.upVotes - props.post.downVotes 
+  const [newVotes,setNewVotes] = React.useState(originalVotes)
 
   const toggleReply = () => setReplyOpen(!replyOpen)
 
+  
+  const upvote = () => {
+      if (newVotes > originalVotes) {
+          setNewVotes(originalVotes)
+          props.onUpvote(false)
+      } else {
```

Comment:
Totally understand that y'all missed this, but there's an edge case this misses for keeping track of total upvotes and downvotes: if the post is currently "downvoted" and you click "upvote," the total upvotes counter will increase by 1, but the total downvotes counter won't decrease!

To fix this, I'd add an extra `if` statement: 
```js
else if (newVotes < originalVotes) { 
  setNewVotes(originalVotes + 1)
  props.onDownvote(false) // reduce total downvotes by 1
  props.onUpvote(true) // increase total upvotes by 1
}
```